gateway/src/main/java/org/georchestra/gateway/security/ResolveHttpHeadersGeorchestraUserFilter.java

```diff
@@ -0,0 +1,145 @@
+                                longname.split(" ")[1], email, "", "", "");
+                        newAccount.setPending(false);
+                        newAccount.setOrg(org);
+                        accountDao.insert(newAccount);
+                        roleDao.addUser(Role.USER, newAccount);
+                        userOpt = map(username);
+                    } catch (DuplicatedUidException e) {
+                        throw new IllegalStateException(e);
+                    } catch (DuplicatedEmailException e) {
+                        throw new IllegalStateException(e);
+                    } catch (DataServiceException e) {
+                        throw new IllegalStateException(e);
+                    }
+                }
+                GeorchestraUsers.store(exchange, userOpt.orElse(null));
+            } else {
+                GeorchestraUsers.store(exchange, map(exchange).orElse(null));
+            }
+            return chain.filter(exchange);
+        } else {
+            return chain.filter(exchange);
+        }
+    }
+
+    protected Optional<GeorchestraUser> map(String username) {
+        UserMapperImpl mapper = new UserMapperImpl();
+        mapper.setRoleDao(roleDao);
+        List<String> protectedUsers = Collections.emptyList();
+        UserRule rule = new UserRule();
+        rule.setListOfprotectedUsers(protectedUsers.toArray(String[]::new));
+        UsersApiImpl usersApi = new UsersApiImpl();
+        usersApi.setAccountsDao(accountDao);
+        usersApi.setMapper(mapper);
+        usersApi.setUserRule(rule);
+
+        Optional<GeorchestraUser> userOpt = usersApi.findByUsername(username);
+        if (userOpt.isPresent()) {
+            List<String> roles = userOpt.get().getRoles().stream().map(r -> r.contains("ROLE_") ? r : "ROLE_" + r)
+                    .collect(Collectors.toList());
+            if (roles.isEmpty()) {
+                roles.add("ROLE_USER");
+            }
+            userOpt.get().setRoles(roles);
+        }
+        return userOpt;
+    }
+
+    protected Optional<GeorchestraUser> map(ServerWebExchange exchange) {
+        String username = exchange.getRequest().getHeaders().get("sec-username").get(0);
+        String email = exchange.getRequest().getHeaders().get("sec-email").get(0);
+        String givenName = exchange.getRequest().getHeaders().get("sec-longname").get(0);
+        String org = exchange.getRequest().getHeaders().get("sec-org").get(0);
+        GeorchestraUser user = new GeorchestraUser();
+        user.setUsername(username);
+        user.setEmail(email);
+        user.setFirstName(givenName);
+        user.setOrganization(org);
+        List<String> roles = user.getRoles().stream().map(r -> r.contains("ROLE_") ? r : "ROLE_" + r)
+                .collect(Collectors.toList());
+        if (roles.isEmpty()) {
+            roles.add("ROLE_USER");
+        }
```

Comment:
I am wondering if not connected users should have the ROLE_USER (as the roles collection is likely to be empty if we are not connected).